packages/deploy/src/utils.ts

```diff
@@ -23,34 +23,59 @@ export function assertBeaconProxy(
 }
 
 export class CheckList {
+  prefix: string; // prefix for all messages in this checklist
   ok: string[]; // successful items
-  error: [string, unknown][]; // failed items with associated error from chai assertion or plain error
-  constructor() {
+  error: unknown[]; // failed items with associated error from chai assertion or plain error
+
+  constructor(prefix: string | void) {
+    this.prefix = prefix ? prefix : '';
     this.ok = [];
     this.error = [];
   }
 
+  static combine(lists: CheckList[]): CheckList {
+    const combinedList = new CheckList();
+    lists.map((list) => {
+      combinedList.ok.push(...list.ok);
+      combinedList.error.push(...list.error);
+    });
+    return combinedList;
+  }
+
+  output(): void {
+    // TODO: improve output readability
+    console.log('Failed: ', JSON.stringify(this.error, null, 2));
```

Comment:
This will output the raw `ethers` error as we see now? Can we perhaps handle that better by output + debug level?

Reply:
Since we are tackling checks, I would really want us to move away from printing that monstrosity 😅 that spams the terminal output